### DISCUSSION TITLE
Docs understate impact of not caching prisma generate

### DISCUSSION THREAD

**Discussion by sepehr500** [vercel/turborepo#5581] [Help] [answered]
It's possible i'm misunderstanding turborepo usage, but the documentation notes `Since prisma generate usually only takes 5-6 seconds, and tends not to take longer with larger schema files, this seems like a fine trade-off.` But since `prisma generate` is often so early in the build chain, the fact you can not cache it means you can almost not cache anything in your build. Even in the example [docs](https://turbo.build/repo/docs/handbook/tools/prisma#6-figuring-out-the-scripts), build depends on `db:generate` so build can never be cached. If you had test and deploy as steps that ran after as well, they would not be able to be cached either. Am I correct in this assessment?

**Comment by anthonyshew** [comment, marked as answer]
Prisma comes with a lot of baggage that makes its usage in monorepos difficult. Our doc is up to date for the best practices that we've found but, if you've found a better strategy, we'd love a PR. 🙏 

As far as your question goes, the build will still be cached. `prisma generate` would be uncached but not the rest of the tasks required to create your build.
